lib/modules/datasource/packagist/schema.ts

```diff
@@ -14,14 +14,54 @@ export const ComposerRelease = z.object({
     .transform((x) => x.url)
     .nullable()
     .catch(null),
+  require: z
+    .object({
+      php: z.string(),
+    })
+    .transform((x) => x.php)
+    .nullable()
+    .catch(null),
   time: z.string().nullable().catch(null),
 });
+type ComposerRelease = z.infer<typeof ComposerRelease>;
+
+type MinifiedRelease = Omit<ComposerRelease, 'version' | 'time'>;
+
+// Ported from: https://github.com/composer/metadata-minifier/blob/main/src/MetadataMinifier.php#L17
+function expandReleases(releases: ComposerRelease[]): ComposerRelease[] {
+  let prevReleaseVals: MinifiedRelease | null = null;
+
+  for (const composerRelease of releases) {
+    if (!prevReleaseVals) {
+      prevReleaseVals = {
+        homepage: composerRelease.homepage,
+        source: composerRelease.source,
+        require: composerRelease.require,
+      };
+      continue;
+    }
+
+    for (const k of Object.keys(prevReleaseVals)) {
+      const key: keyof MinifiedRelease = k as keyof MinifiedRelease;
+      const prevReleaseVal = prevReleaseVals[key];
+      const currentReleaseVal = composerRelease[key];
+      if (currentReleaseVal !== null) {
+        prevReleaseVals[key] = currentReleaseVal;
+      } else if (prevReleaseVal !== null) {
+        composerRelease[key] = prevReleaseVal;
+      }
```

Comment:
Not sure this is the same as the `__unset` logic in composer?

Reply:
I pulled https://3v4l.org/IjoG6L out of their test suite, maybe this helps you verify the logic. It prints a minified and an expanded array with the exact same expand logic as in the lib